daemon/config/config.go

```diff
@@ -351,8 +352,21 @@ func getConflictFreeConfiguration(configFile string, flags *pflag.FlagSet) (*Con
 	}
 
 	reader = bytes.NewReader(b)
-	err = json.NewDecoder(reader).Decode(&config)
-	return &config, err
+	if err := json.NewDecoder(reader).Decode(&config); err != nil {
```

Comment:
Looks like validation doesn't currently catch specifying both `--graph` and `--data-dir` as flags